Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -290,8 +290,11 @@ public InterpreterScope(AnalysisValue av, InterpreterScope outerScope)
             _nodeScopes.Clear();
         }
 
-        public virtual IAnalysisSet AddNodeValue(Node node, IAnalysisSet variable) {
-            return _nodeValues[node] = variable;
+        public virtual IAnalysisSet AddNodeValue(Node node, NodeValueKind kind, IAnalysisSet variable) {
+            NodeValue next;
+            _nodeValues.TryGetValue(node, out next);
+            _nodeValues[node] = new NodeValue(kind, variable, next);
```

Comment:
I'm more worried about hitting a case where we incorrectly keep adding a node value against the same node with the same type, without checking whether it already exists (possibly some time in the future when someone else calls this). Maybe making it a GetOrAddNodeValue() function would address that?